src/tv2-common/actions/executeAction.ts

```diff
@@ -1109,6 +1110,32 @@ async function executePiece<
 	}
 }
 
+function isPlannedLivePiece(currentPiece: IBlueprintPieceInstance<PieceMetaData>, liveSourceLayerId: string) {
+	return (
+		currentPiece.piece.sourceLayerId === liveSourceLayerId &&
+		!currentPiece.piece.metaData?.modifiedByAction &&
+		!currentPiece?.dynamicallyInserted
+	)
+}
+
+function findLastPlayingPieceInstance(
+	currentPieceInstances: Array<IBlueprintPieceInstance<PieceMetaData>>,
+	sourceLayerIds: string[]
+): IBlueprintPieceInstance<PieceMetaData> | undefined {
+	const playingPiecesOnSelectedLayers = currentPieceInstances.filter(
+		(p) => !p.stoppedPlayback && sourceLayerIds.includes(p.piece.sourceLayerId)
+	)
+	if (playingPiecesOnSelectedLayers.length < 2) {
```

Comment:
I think that it's not necessary because the code just read _if array has fewer than 2 elements, return the first one (possibly undefined)_, and it is inferred that this is because calling `reduce` with less than two elements is pointless (and wrong), but I changed it to `<= 1` because it looks a bit nicer